Firestore/src/Query.php

```diff
@@ -321,94 +323,38 @@ public function select(array $fieldPaths)
      * $query = $query->where('friends', 'array-contains', ['Steve', 'Sarah']);
      * ```
      *
-     * @param string|FieldPath $fieldPath The field to filter by.
+     * ```
+     * use Google\Cloud\Firestore\Filter;
+     *
+     * // Filtering with Filter::or
+     * $query = $query->where(Filter::or([Filter::field('firstName', '=', 'John'),
+     *                          Filter::field('firstName', '=', 'Monica')]));
+     * ```
+     *
+     * @param string|array|FieldPath $fieldPath The field to filter by, or array of Filters.
      * @param string|int $operator The operator to filter by.
      * @param mixed $value The value to compare to.
      * @return Query A new instance of Query with the given changes applied.
      * @throws \InvalidArgumentException If an invalid operator or value is encountered.
      */
-    public function where($fieldPath, $operator, $value)
+    public function where($fieldPath, $operator = null, $value = null)
     {
-        $basePath = $this->basePath();
-
-        if ($value instanceof FieldValueInterface) {
-            throw new \InvalidArgumentException(sprintf(
-                'Value cannot be a `%s` value.',
-                FieldValue::class
-            ));
-        }
-
-        if (!($fieldPath instanceof FieldPath)) {
-            $fieldPath = FieldPath::fromString($fieldPath);
-        }
-
-        $escapedPathString = $fieldPath->pathString();
-
-        // alias for friendlier error message below.
-        $originalOperator = $operator;
-        $operator = array_key_exists(strtolower($operator), $this->shortOperators)
-            ? $this->shortOperators[strtolower($operator)]
-            : $operator;
-
-        try {
-            FieldFilterOperator::name($operator);
-        } catch (\UnexpectedValueException $e) {
-            throw new \InvalidArgumentException(sprintf(
-                'Operator %s is not a valid operator',
-                $originalOperator
-            ));
-        }
-
-        if ($escapedPathString === self::DOCUMENT_ID) {
-            if ($operator === FieldFilterOperator::IN) {
-                $value = array_map(function ($value) use ($basePath) {
-                    return $this->createDocumentReference($basePath, $value);
-                }, (array) $value);
-            } else {
-                $value = $this->createDocumentReference($basePath, $value);
-            }
-        }
-
-        if ((is_float($value) && is_nan($value)) || is_null($value)) {
-            if ($operator !== FieldFilterOperator::EQUAL) {
-                throw new \InvalidArgumentException('Null and NaN are allowed only with operator EQUALS.');
+        if (is_array($fieldPath)) {
+            if (!is_null($operator) or !(is_null($operator))) {
+                throw new \InvalidArgumentException(
+                    'Filter with operator and value is not accepted'
+                );
             }
```

Comment:
This is not needed, you can add it in PhpDoc that in case filter array is provided, other params will be ignored.

Reply:
Done!